device/dell/x86_64-dell_s6000_s1220-r0/plugins/sfputil.py

```diff
@@ -175,9 +194,42 @@ def reset(self, port_num):
         return True
 
     def get_transceiver_change_event(self):
```

Comment:
i thought we should do some sleep, otherwise it becomes busy polling and will cause high cpu utilization.

Reply:
- Guohan, I introduced a 1 second sleep to-avoid-busy loop. Even, if any state change happens within that short window, the fix check the state change and update to xcvrd accordingly.
- Introduced get_transceiver_status() fucntion for reading  qsfp_modprs file

Reply:
get_transceiver_change_event is a blocking call, you should sleep in your while loop.